datafusion/physical-expr/src/functions.rs

```diff
@@ -191,9 +191,11 @@ pub(crate) enum Hint {
     AcceptsSingular,
 }
 
-/// decorates a function to handle [`ScalarValue`]s by converting them to arrays before calling the function
+/// Decorates a function to handle [`ScalarValue`]s by converting them to arrays before calling the function
 /// and vice-versa after evaluation.
-pub fn make_scalar_function<F>(inner: F) -> ScalarFunctionImplementation
+/// Note that this function makes a scalar function with no arguments or all scalar inputs return a scalar.
+/// That's said its output will be same for all input rows in a batch.
+pub(crate) fn make_scalar_function<F>(inner: F) -> ScalarFunctionImplementation
```

Comment:
Except for the usage in same crate, its other usages are in tests.

Reply:
For better or worse, since this function is in the `example_udf.rs` file, I think it is likely to be widely used when people followed the example model.

Thus I think this change is likely to be disruptive to exising users.

To help soften the blow, I agree that deprecating it for a release or two would be the best

Perhaps we could do something like

```suggestion
#[deprecated(since = "35.0.0", note = "Implement your function directly in terms of ColumnarValue or use `ScalarUDF` instead")]
pub fn make_scalar_function<F>(inner: F) -> ScalarFunctionImplementation
  make_scaler_function_inner(inner)
}

/// Internal implementation, see comments on `make_scalar_function` for caveats
pub(crate) fn make_scalar_function_inner<F>(inner: F) -> ScalarFunctionImplementation 

```